### PR TITLE
V2 v1 redirect

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
When a user starts a v1 or v2 form, continuing the form from either intro page will take them to the appropriate form. For example, if I have a v1 form saved in progress, and I hit continue from the v2 intro page, it'll take me to where I left off in the v1 app. The reverse is also true.

## Testing done
All manual testing. Not sure how to write tests for this, but it's kind of a hack, so :man_shrugging:

## Screenshots
N/A

## Acceptance criteria
- [x] The user is directed to the the form where they left off from either the v1 or v2 intro pages

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/save-in-progress/RoutedSavableApp.jsx [hunk 1]
```diff
@@ -90,7 +90,29 @@ class RoutedSavableApp extends React.Component {
     ) {
       newProps.router.replace(newProps.returnUrl);
     } else if (status === LOAD_STATUSES.success) {
-      newProps.router.push(newProps.returnUrl);
+      // TODO: Remove this when 526 v1 is depreciated
```
Comment (jbalboni) [inline src/platform/forms/save-in-progress/RoutedSavableApp.jsx]: Would it cause much trouble to make an `onFormLoaded` hook in the form config that was called here and move all of this logic out of RoutedSavableApp and into the form specific code? Even if this is really short term, it's still 526 specific code in our site wide SiP components.

Comment (cvalarida) [reply src/platform/forms/save-in-progress/RoutedSavableApp.jsx]: I dig it! Good thinking. :raised_hands:

**Review by jbalboni — APPROVED** [review]

File: src/platform/forms/save-in-progress/RoutedSavableApp.jsx [hunk 2]
```diff
@@ -90,7 +90,12 @@ class RoutedSavableApp extends React.Component {
     ) {
       newProps.router.replace(newProps.returnUrl);
     } else if (status === LOAD_STATUSES.success) {
-      newProps.router.push(newProps.returnUrl);
+      if (newProps.formConfig.onFormLoaded) {
+        // The onFormLoaded callback should handle navigating to the start of the form
+        newProps.formConfig.onFormLoaded(newProps);
```
Comment (bshyong) [inline src/platform/forms/save-in-progress/RoutedSavableApp.jsx]: 👍

**Review by bshyong — APPROVED** [review]

**Review by markgreenburg — APPROVED** [review]
🚀

**Pull request merged**